### PR TITLE
15 us14 post reviews

### PR DESCRIPTION
**Pull request by kristijonasbu**
Added the create review modal to the placeView component

### CODE REVIEW

File: ms-frontend/src/components/CreateReviewModal.vue [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+//     } catch (error) {
+//         console.error('Error accessing the created reviews:', error)
+//     }
+
+//     try {
+//         await reviewService.updateReviewImage(myReviews.value?.[0].review_id, reviewPicture)
+//     } catch (error) {
+//         console.error('Error updating review image:', error)
+//     }
+// }
```
Comment (AlessioBrazerol) [inline ms-frontend/src/components/CreateReviewModal.vue:223]:
```suggestion
const createReview = async () => {
  if (checkInputValidity()) {
    try {
      // Call createReview method from reviewService
      await reviewService.createReview(reviewText.value, rating.value, props.location).then((response: Review) => {
        updateReviewImage(response.id)
      })
      emit('closeModal')
    } catch (error) {
      console.error('Error creating review:', error)
    }
  }
}

const updateReviewImage = async (reviewId: string) => {
    if(reviewPicture.value === null) return
    try {
        await reviewService.appendReviewImage(reviewId, reviewPicture.value)
    } catch (error) {
        console.error('Error updating review image:', error)
    }
}
```

With this you could update the profile picture after the review was created.

File: ms-frontend/src/services/reviewService.ts [hunk 2]
```diff
@@ -1,22 +1,35 @@
...
+        name: location.name,
+        type: location.type,
+        coordinates: {
+          x: location.coordinates.x,
+          y: location.coordinates.y
+        }
+      }
+    })
+  },
+
```
Comment (AlessioBrazerol) [inline ms-frontend/src/services/reviewService.ts:22]:
```suggestion
    async appendReviewImage(reviewId: string, file: File) {
    const formData = new FormData()
    formData.append('review_id', reviewId)
    formData.append('image', file)

    const response = await apiClient.patch('reviews/image', formData)
    return new Review(response.data)
  },
```

Add to append a review image.

File: ms-frontend/src/components/CreateReviewModal.vue [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+import { reviewService } from '@/services/reviewService'
+import Textarea from 'primevue/textarea'
+import type { User } from '@/types/User'
+import type { Location } from '@types/Location'
+import type { Review } from '@types/Review'
+
+const authStore = useAuthStore()
+
+const signedInUser = authStore.user
+const myId = signedInUser?.id ?? ''
```
Comment (AlessioBrazerol) [inline ms-frontend/src/components/CreateReviewModal.vue:135]:
```suggestion
import { ref, watch } from 'vue'
import FileUpload from '@/components/FileUpload.vue'
import { reviewService } from '@/services/reviewService'
import Textarea from 'primevue/textarea'
import type { Location } from '@types/Location'
import type { Review } from '@types/Review'

```

Remove unused imports.

File: ms-frontend/src/icons/UploadIcon.vue
Comment (AlessioBrazerol) [inline ms-frontend/src/icons/UploadIcon.vue:1]: This icon is never used and probably the same as 'arrow-up-tray'.
I would remove it.

**Review by AlessioBrazerol — CHANGES_REQUESTED** [review]
Generally looks good and works really well.
I added some change requests, so image upload also works.

File: ms-frontend/src/services/reviewService.ts [hunk 4]
```diff
@@ -19,7 +19,14 @@ export const reviewService = {
       }
     })
```
Comment (AlessioBrazerol) [inline ms-frontend/src/services/reviewService.ts:20]: `const response = await apiClient.pos(....

}
return new Review(response.data)
`

You need to add this to the createReview method too, sorry forgot to mention it.

**Review by AlessioBrazerol — CHANGES_REQUESTED** [review]

File: ms-frontend/src/components/CreateReviewModal.vue (hunk 1 above)
Comment (kristijonasbu) [reply ms-frontend/src/components/CreateReviewModal.vue]: This gives me 'Request failed with status code 400', trying to look at it now

Comment (AlessioBrazerol) [reply ms-frontend/src/components/CreateReviewModal.vue]: Did you add this [Change](https://github.com/uzh-network-ninjas/ase-social-network/pull/153/files#r1582062079) ?

Comment (kristijonasbu) [reply ms-frontend/src/components/CreateReviewModal.vue]: yup, still gives the 400 bad request error:/

**Review by AlessioBrazerol — APPROVED** [review]

**Pull request merged**